test/common/http/http2/conn_pool_test.cc

```diff
@@ -1434,6 +1434,49 @@ TEST_F(Http2ConnPoolImplTest, PreconnectWithoutMultiplexing) {
   closeAllClients();
 }
 
+TEST_F(Http2ConnPoolImplTest, IncreaseCapacityWithSettingsFrame) {
```

Comment:
it would be good to integration test this as well.
I think you could move TEST_P(QuicHttpIntegrationTest, NoInitialStreams) { to the protocol integration test, and updateConcurrentStreams / sendSettings based on HTTP/3 vs HTTP/3?

Reply:
If I understand correctly, do you mean move `NoInitialStreams` to ``protocol_integration_test.cc`` and let it test both HTTP2 and HTTP3?  It seems a bit complicated, could I open another PR to implement it after dig a little deeper :)

Reply:
I'd definitely like to make sure we integration test incrementing settings with this change.  Alternately you can land this false by default with a TODO to flip true with integration testing, but I don't think adapting the test is too complicated so I think it makes more sense to land the test now.  I'm happy to help if you get stuck!

Reply:
Get it, let me do some research to get familiar with the integration test framework :)